### PR TITLE
Add taiga style 1.12

### PR DESCRIPTION
**Pull request by Raycoms**
TheSwedishPilot made a nice taiga style for us

Review please

### CODE REVIEW

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 32 issues
* ![MINOR][MINOR] 32 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/42a1a658eb10d54ad1479745f405b140458941d7/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/42a1a658eb10d54ad1479745f405b140458941d7/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/42a1a658eb10d54ad1479745f405b140458941d7/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/42a1a658eb10d54ad1479745f405b140458941d7/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/42a1a658eb10d54ad1479745f405b140458941d7/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/42a1a658eb10d54ad1479745f405b140458941d7/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [requestsystem](https://github.com/Minecolonies/minecolonies/blob/42a1a658eb10d54ad1479745f405b140458941d7/src/main/java/com/minecolonies/coremod/colony/requestsystem): Add a 'package-info.java' file to document the 'requestsystem' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [init](https://github.com/Minecolonies/minecolonies/blob/42a1a658eb10d54ad1479745f405b140458941d7/src/main/java/com/minecolonies/coremod/colony/requestsystem/init): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [locations](https://github.com/Minecolonies/minecolonies/blob/42a1a658eb10d54ad1479745f405b140458941d7/src/main/java/com/minecolonies/coremod/colony/requestsystem/locations): Add a 'package-info.java' file to document the 'locations' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [requests](https://github.com/Minecolonies/minecolonies/blob/42a1a658eb10d54ad1479745f405b140458941d7/src/main/java/com/minecolonies/coremod/colony/requestsystem/requests): Add a 'package-info.java' file to document the 'requests' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 32 issues
* ![MINOR][MINOR] 32 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/012fa6c066c8bb41a3303ebfb29e2bbb35ec61b4/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/012fa6c066c8bb41a3303ebfb29e2bbb35ec61b4/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/012fa6c066c8bb41a3303ebfb29e2bbb35ec61b4/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/012fa6c066c8bb41a3303ebfb29e2bbb35ec61b4/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/012fa6c066c8bb41a3303ebfb29e2bbb35ec61b4/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/012fa6c066c8bb41a3303ebfb29e2bbb35ec61b4/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [requestsystem](https://github.com/Minecolonies/minecolonies/blob/012fa6c066c8bb41a3303ebfb29e2bbb35ec61b4/src/main/java/com/minecolonies/coremod/colony/requestsystem): Add a 'package-info.java' file to document the 'requestsystem' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [init](https://github.com/Minecolonies/minecolonies/blob/012fa6c066c8bb41a3303ebfb29e2bbb35ec61b4/src/main/java/com/minecolonies/coremod/colony/requestsystem/init): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [locations](https://github.com/Minecolonies/minecolonies/blob/012fa6c066c8bb41a3303ebfb29e2bbb35ec61b4/src/main/java/com/minecolonies/coremod/colony/requestsystem/locations): Add a 'package-info.java' file to document the 'locations' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [requests](https://github.com/Minecolonies/minecolonies/blob/012fa6c066c8bb41a3303ebfb29e2bbb35ec61b4/src/main/java/com/minecolonies/coremod/colony/requestsystem/requests): Add a 'package-info.java' file to document the 'requests' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Asherslab** [comment]
Looks great! :shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1603/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 40 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 33 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [PlacementHandlers.java#L74](https://github.com/Minecolonies/minecolonies/blob/43f54e9cda63a8237b1bd29e5a30fe590b560974/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L74): Reduce this class from 36 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L263](https://github.com/Minecolonies/minecolonies/blob/43f54e9cda63a8237b1bd29e5a30fe590b560974/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L263): Reduce this class from 39 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L307](https://github.com/Minecolonies/minecolonies/blob/43f54e9cda63a8237b1bd29e5a30fe590b560974/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L307): Reduce this class from 32 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L394](https://github.com/Minecolonies/minecolonies/blob/43f54e9cda63a8237b1bd29e5a30fe590b560974/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L394): Reduce this class from 37 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L436](https://github.com/Minecolonies/minecolonies/blob/43f54e9cda63a8237b1bd29e5a30fe590b560974/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L436): Reduce this class from 36 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L478](https://github.com/Minecolonies/minecolonies/blob/43f54e9cda63a8237b1bd29e5a30fe590b560974/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L478): Reduce this class from 32 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L513](https://github.com/Minecolonies/minecolonies/blob/43f54e9cda63a8237b1bd29e5a30fe590b560974/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L513): Reduce this class from 36 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/43f54e9cda63a8237b1bd29e5a30fe590b560974/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/43f54e9cda63a8237b1bd29e5a30fe590b560974/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/43f54e9cda63a8237b1bd29e5a30fe590b560974/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java [hunk 1]
```diff
@@ -251,8 +251,11 @@ private void addHomelessCitizens()
             // Move the citizen to a better hut
             if (citizen.getHomeBuilding() != null && citizen.getHomeBuilding().getBuildingLevel() < this.getBuildingLevel())
             {
-                // The citizen can move to this hut to improve conditions
-                citizen.getHomeBuilding().removeCitizen(citizen);
+                if(citizen.getHomeBuilding() instanceof BuildingHome)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java:254]: MAJOR Merge this if statement with the enclosing one. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 46 issues
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 37 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BuildingHome.java#L358](https://github.com/Minecolonies/minecolonies/blob/7ee760b32e852f850552b272056e6bb2d8ada2ce/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java#L358): Reduce this class from 38 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L74](https://github.com/Minecolonies/minecolonies/blob/7ee760b32e852f850552b272056e6bb2d8ada2ce/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L74): Reduce this class from 36 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L263](https://github.com/Minecolonies/minecolonies/blob/7ee760b32e852f850552b272056e6bb2d8ada2ce/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L263): Reduce this class from 39 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L307](https://github.com/Minecolonies/minecolonies/blob/7ee760b32e852f850552b272056e6bb2d8ada2ce/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L307): Reduce this class from 32 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L394](https://github.com/Minecolonies/minecolonies/blob/7ee760b32e852f850552b272056e6bb2d8ada2ce/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L394): Reduce this class from 37 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L436](https://github.com/Minecolonies/minecolonies/blob/7ee760b32e852f850552b272056e6bb2d8ada2ce/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L436): Reduce this class from 36 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L478](https://github.com/Minecolonies/minecolonies/blob/7ee760b32e852f850552b272056e6bb2d8ada2ce/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L478): Reduce this class from 32 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L513](https://github.com/Minecolonies/minecolonies/blob/7ee760b32e852f850552b272056e6bb2d8ada2ce/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L513): Reduce this class from 36 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/7ee760b32e852f850552b272056e6bb2d8ada2ce/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/7ee760b32e852f850552b272056e6bb2d8ada2ce/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java [hunk 2]
```diff
@@ -545,4 +552,51 @@ else if (!world.setBlockState(pos, blockState, UPDATE_FLAG))
             return blockState;
         }
     }
+
+    public static class RackPlacementHandler implements IPlacementHandler
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java:556]: MAJOR Reduce this class from 41 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 47 issues
* ![MAJOR][MAJOR] 10 major
* ![MINOR][MINOR] 37 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BuildingHome.java#L358](https://github.com/Minecolonies/minecolonies/blob/19418e64c74e90b441de6bb57ac2c0470114bee5/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java#L358): Reduce this class from 38 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L76](https://github.com/Minecolonies/minecolonies/blob/19418e64c74e90b441de6bb57ac2c0470114bee5/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L76): Reduce this class from 36 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L265](https://github.com/Minecolonies/minecolonies/blob/19418e64c74e90b441de6bb57ac2c0470114bee5/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L265): Reduce this class from 39 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L309](https://github.com/Minecolonies/minecolonies/blob/19418e64c74e90b441de6bb57ac2c0470114bee5/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L309): Reduce this class from 32 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L396](https://github.com/Minecolonies/minecolonies/blob/19418e64c74e90b441de6bb57ac2c0470114bee5/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L396): Reduce this class from 37 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L438](https://github.com/Minecolonies/minecolonies/blob/19418e64c74e90b441de6bb57ac2c0470114bee5/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L438): Reduce this class from 36 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L480](https://github.com/Minecolonies/minecolonies/blob/19418e64c74e90b441de6bb57ac2c0470114bee5/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L480): Reduce this class from 32 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [PlacementHandlers.java#L515](https://github.com/Minecolonies/minecolonies/blob/19418e64c74e90b441de6bb57ac2c0470114bee5/src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java#L515): Reduce this class from 36 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/19418e64c74e90b441de6bb57ac2c0470114bee5/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/19418e64c74e90b441de6bb57ac2c0470114bee5/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java [hunk 3]
```diff
@@ -3,6 +3,7 @@
 import com.minecolonies.api.util.CompatibilityUtils;
 import com.minecolonies.api.util.InventoryUtils;
 import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.api.util.constant.Constants;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java:6]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Constants'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 99 issues
* ![MAJOR][MAJOR] 31 major
* ![MINOR][MINOR] 66 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractBuilding.java#L54](https://github.com/Minecolonies/minecolonies/blob/c99498828b38b57b4c50a75f721163a6042f844b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L54): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 40 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L54](https://github.com/Minecolonies/minecolonies/blob/c99498828b38b57b4c50a75f721163a6042f844b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L54): class "AbstractBuilding" has 68 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L1345](https://github.com/Minecolonies/minecolonies/blob/c99498828b38b57b4c50a75f721163a6042f844b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1345): Reduce this class from 75 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [AbstractBuildingGuards.java#L569](https://github.com/Minecolonies/minecolonies/blob/c99498828b38b57b4c50a75f721163a6042f844b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java#L569): Reduce this class from 58 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [AbstractBuildingWorker.java#L170](https://github.com/Minecolonies/minecolonies/blob/c99498828b38b57b4c50a75f721163a6042f844b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java#L170): Either log or rethrow this exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AbstractBuildingWorker.java#L326](https://github.com/Minecolonies/minecolonies/blob/c99498828b38b57b4c50a75f721163a6042f844b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java#L326): Reduce this class from 52 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [BuildingFarmer.java#L538](https://github.com/Minecolonies/minecolonies/blob/c99498828b38b57b4c50a75f721163a6042f844b/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingFarmer.java#L538): Reduce this class from 86 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [BuildingHome.java#L358](https://github.com/Minecolonies/minecolonies/blob/c99498828b38b57b4c50a75f721163a6042f844b/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java#L358): Reduce this class from 38 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [BuildingLumberjack.java#L217](https://github.com/Minecolonies/minecolonies/blob/c99498828b38b57b4c50a75f721163a6042f844b/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L217): Reduce this class from 73 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [BuildingMiner.java#L458](https://github.com/Minecolonies/minecolonies/blob/c99498828b38b57b4c50a75f721163a6042f844b/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingMiner.java#L458): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 4]
```diff
@@ -240,6 +240,11 @@
     private double overallHappiness = 5;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java:240]: MAJOR Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 130 issues
* ![MAJOR][MAJOR] 49 major
* ![MINOR][MINOR] 76 minor
* ![INFO][INFO] 5 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/307adb9c65657305a7a650f0c144e07d71f923e0/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.387 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L55](https://github.com/Minecolonies/minecolonies/blob/307adb9c65657305a7a650f0c144e07d71f923e0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 50 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L55](https://github.com/Minecolonies/minecolonies/blob/307adb9c65657305a7a650f0c144e07d71f923e0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L55): class "Colony" has 100 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L371](https://github.com/Minecolonies/minecolonies/blob/307adb9c65657305a7a650f0c144e07d71f923e0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L371): Remove this useless assignment to local variable "achievementKey". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [Colony.java#L539](https://github.com/Minecolonies/minecolonies/blob/307adb9c65657305a7a650f0c144e07d71f923e0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L539): Remove this useless assignment to local variable "achievementCompound". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [Colony.java#L660](https://github.com/Minecolonies/minecolonies/blob/307adb9c65657305a7a650f0c144e07d71f923e0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L660): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1109](https://github.com/Minecolonies/minecolonies/blob/307adb9c65657305a7a650f0c144e07d71f923e0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1109): The Cyclomatic Complexity of this method "onWorldTick" is 20 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1293](https://github.com/Minecolonies/minecolonies/blob/307adb9c65657305a7a650f0c144e07d71f923e0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1293): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1295](https://github.com/Minecolonies/minecolonies/blob/307adb9c65657305a7a650f0c144e07d71f923e0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1295): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1297](https://github.com/Minecolonies/minecolonies/blob/307adb9c65657305a7a650f0c144e07d71f923e0/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1297): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**